File: go/arrays_and_hashing/longest_consecutive_sequence.go
package main

func longestConsecutive(nums []int) int {
    numSet := map[int]bool{}
    longest := 0

    for _, num := range nums {
        numSet[num] = true
    }

    for _, num := range nums {
        _, exists := numSet[num - 1]
        if !exists {
            length := 1
            for numSet[num + length] {
                length++
            }
            if length > longest {
                longest = length
            }
        }
    }

    return longest
}

File: python/arrays_and_hashing/longest_consecutive_sequence.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        nums_set = set(nums)

        for num in nums_set:
            if num - 1 not in nums_set:
                length = 1
                while num + length in nums_set:
                    length += 1

                longest = max(longest, length)

        return longest
